third_party/txt/src/minikin/FontCollection.cpp

```diff
@@ -528,8 +529,27 @@ void FontCollection::itemize(const uint16_t* string,
           }
           start -= prevChLength;
         }
-        result->push_back(
-            {family->getClosestMatch(style), static_cast<int>(start), 0});
+        result->push_back({family->getClosestMatch(style, ch, variant),
+                           static_cast<int>(start), 0});
+        run = &result->back();
+        lastFamily = family.get();
+      } else if (family.get() == lastFamily &&
+                 // We need to change this condition to more general logic. And
+                 // when changing the font, it is necessary to check whether the
+                 // existing characters also have glyphs.
```

Comment:
Yes, I agree too and I think font class must provide a function to check the presence of glyphs.